spec/helpers/ui_helper_spec.rb

```diff
@@ -23,5 +23,86 @@
       end
     end
   end
+
+  describe 'input_add_button' do
+    context "with default options" do
+      subject { helper.input_add_button("Label", container_selector: "Container") { "Block" } }
+
+      it 'should generate a button with correct attributes' do
+        unsafe_html = CGI.unescapeHTML(subject)
+
+        expect(unsafe_html).to match(/<div>/)
+        expect(unsafe_html).to match(/<a.* class="btn btn-outline-primary".*>/)
+        expect(unsafe_html).to match(/<a.* data-form-input-target="addButton".*>/)
+        expect(unsafe_html).to match(/<a.* data-add-dest-selector="Container".*>/)
+        expect(unsafe_html).to match(/<a.* data-action="click->form-input#addItem:prevent".*>/)
+        expect(unsafe_html).to match(/<a.*>Label<\/a>/)
+        expect(unsafe_html).to match(/<template.* data-form-input-target="addTemplate".*>/)
+        expect(unsafe_html).to match(/<template.*>Block<\/template>/)
+      end
+    end
+
+    context "with custom options" do
+      subject {
+        helper.input_add_button("Label", container_selector: "Container", class: "Class", id: "Id",
+          data: {test: "test"}) { "Block" }
+      }
+
+      it 'should generate a button with correct attributes' do
+        unsafe_html = CGI.unescapeHTML(subject)
+
+        expect(unsafe_html).to match(/<div>/)
```

Comment:
Is there a reason you can't just compare the HTML directly against a string? If not, you can instead use Nokogiri to turn it into a document and you can use CSS or XPath expectations - it's easier and less brittle than regex.

Reply:
> Is there a reason you can't just compare the HTML directly against a string? 

There were a bunch of escape characters that I didn't want to add to the string. 

> Nokogiri to turn it into a document and you can use CSS or XPath expectations - it's easier and less brittle than regex.

Was not aware of this option. I've never written a "helper" spec before so I honestly had not clue what assertions were available, I was just riffing off what was already there.